### ISSUE TITLE
CA1062 is reported strangely when using conditional operator

### ISSUE THREAD

**Issue by Tan90909090** [dotnet/roslyn-analyzers#2280]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.1

#### Diagnostic ID

[CA1062](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1062-validate-arguments-of-public-methods)

#### Repro steps
```cs
using System;

public class Node
{
    public Node Parent { get; set; }

    public static Node M(Node node, bool flag)
    {
        if (node == null) { throw new ArgumentNullException(nameof(node)); }

        var parent = flag ? node.Parent : node;
        return parent.Parent; // CA1062 is reported on `parent`
    }
}
```

#### Expected behavior
CA1062 is not reported.
"parent may refer to null" may be reported.

#### Actual behavior
CA1062 is reported:
`In externally visible method 'Node Node.M(Node node, bool flag)', validate parameter 'node' is non-null before using it. If appropriate, throw an ArgumentNullException when the argument is null or add a Code Contract precondition asserting non-null argument.`

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/2345
